crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1270,16 +1270,53 @@ fn handle_import_from_comment<'a>(
 fn handle_with_comment<'a>(
     comment: DecoratedComment<'a>,
     with_statement: &'a ast::StmtWith,
+    locator: &Locator,
 ) -> CommentPlacement<'a> {
-    if comment.line_position().is_end_of_line()
-        && with_statement.items.first().is_some_and(|with_item| {
-            with_statement.start() < comment.start() && comment.start() < with_item.start()
-        })
+    if comment.line_position().is_own_line() {
+        return CommentPlacement::Default(comment);
+    }
+
+    if with_statement.items.first().is_some_and(|with_item| {
+        with_statement.start() < comment.start() && comment.start() < with_item.start()
+    }) {
+        return CommentPlacement::dangling(comment.enclosing_node(), comment);
+    }
```

Comment:
Nit: We could integrate this into the `if let Some(...)` below

```rust
if let Some(AnyNodeRef::WithItem(...) {
	
} else if let Some(following) = comment.following_node() {
	// First with item
	if comment.start() < with_item.start() {
		return CommentPlacement::dangling(comment.enclosing_node(), comment);
	}
}

CommentPlacement::Default(comment)
```

Reply:
Doesn't the following node need to be the first `WithItem` though?

Reply:
Isn't this guaranteed by preceding node being `None`?

Reply:
Hmm... I think there are some issues with the ranges on these `WithItem`s. Need to look into it.

Reply:
Okay, this finally works :joy: